File: Programmers/src/N개의최소공배수.java
class Solution {

    public int solution(int[] arr) {
        int num = lcm(arr[0], arr[1]);
        int answer = 0;
        
        for(int i=2; i<arr.length; i++){
            answer = lcm(num, arr[i]);
            num = answer;
        }
        
        return answer;
    }
    
    public int gcd(int a, int b) {
        while(b!=0){
            int r = a%b;
            a=b;
            b=r;
        }
        
        return a;
    }
    
    public int lcm(int a, int b) {
        return (a*b)/gcd(a,b);
    }
}

File: Programmers/src/Javascript/Lv2/N개의최소공배수.js
function solution(arr) {
    let answer = 1;
    
    for(let i=0; i<arr.length; i++) 
        answer = lcm(answer, arr[i]);
    
    return answer;
}

// 최대 공약수
const gcd = (a,b) => {
    while(b>0) {
        let temp = b;
        b = a % b;
        a = temp;
    }
    
    return a;
}

// 최소 공배수
const lcm = (a,b) => {
    return (a*b) / gcd(a,b);
}